modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/fsm/DdlBatchingHelper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.exec.fsm;
+
+import org.apache.calcite.sql.SqlNode;
+import org.apache.ignite.internal.sql.engine.sql.ParsedResult;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ *  Provide helper methods for batched DDL commands.
+ */
+public class DdlBatchingHelper {
+    /**
+     * Returns {@code true} if given statement is compatible with script statement.
+     * Node: the operation is not commutative.
+     */
+    static boolean isCompatible(ParsedResult scriptStatement, ParsedResult statement) {
```

Comment:
tbh parameter names are confusing. In `MultiStatementHandler` we pass `currentStatement` and `nextStatement` as input, but semantically we need to check if two arbitrary statements are compatible. 

Let's rename to something like thisStatement-thatStatement or statement1-statement2 or whatever neutral. Similar with batchGroup